rfcs/0098-community-team.md

```diff
@@ -0,0 +1,190 @@
+
+  In exceptional circumstances it is permissible for team members not directly involved in the proceedings to miss them, but these authors do consider this ceremony to be an important ritual for establishing a coherent culture, and hope that team members will keep that in mind and make every effort to attend.
+  
+  The ceremony will be led by an existing team member who will introduce newcomers and announce retirements. The member leading the ceremony will explain the team's values and responsibilities, and how the community team operates. New members will then each, individually state: "I understand my responsibilities as a member of the community team and agree to uphold the community values." Following the formal ceremony, there will be celebratory social interaction.
+
+  This ceremony may seem extremely formal for a technical community, but it serves an important purpose. In a world with very little respect for ceremony, it's important to have things that bring us together, even if they are simple things. Since the community team's responsibility is an important one, it is appropriate for team members to treat it with gravity. It is likewise important to take time to celebrate happy things, so that shared experiences can become the basis of cooperation.
+
+## Authority of the community team
+[authority]: #authority
+
+Team members will have the authority to ban community members from any community social space, or from all such spaces. Team members will be added to all necessary access control lists to exercise this power, as soon as possible after the onboarding ceremony.
+
+Team members will manage their own access control lists, adding and removing each other on Discourse, GitHub, Mjolnir (the Matrix bot), and anywhere else that's necessary.
+
+Community social spaces include Disourse, GitHub, and Matrix, as well as any analogous spaces which may exist in the future. It is expected that team members will not act alone when implementing bans, except in situations which are both urgent and exceptional. Like all moderation actions, bans will be publicly disclosed to the community, and explained. Banning somebody from participation on GitHub does prevent them from submitting pull requests; it is thus equivalent to full expulsion from the technical effort which the community is dedicated to.
+
+It is necessary for the team to have the authority to ban, as it is the basis of all the soft power the team wields. However, the team should prefer to use bans only as a last resort. When somebody is excluded from a community, the community has no further power to change their heart, help them grow, or otherwise influence them. Bans should not be performed vindictively or as punishments, but only as necessary actions for the well being of the remaining community members. A ban is a sad occasion, no matter how necessary it may be.
```

Comment:
Shouldn't likely loss of ability to collaborate also be considered a drawback of a ban?

Some communities have policies like starting with temporary bans, and sometimes banning a human for more than a chosen single-digit number of years requires truly exceptional justification and confidence, even by the standards of banning. Should there be any recommendations along these lines?

Reply:
The goal here is not to make specific rules or recommendations, because anything like that which is specified in a founding document will be taken, by some people, as permission to violate the social norms as long as they're willing to accept the penalty. The proposal is that the community team will make those practical decisions based on context, as needed.

I do agree with you that not being able to collaborate with somebody is also a drawback of a ban. This wording is mostly meant to appeal to anyone who might otherwise make a philosophical defense of the importance of bans, so it's couched in terms that I hope will address those arguments.

The RFC's goal in referencing bans is to bring them on the table as a possible means of resolving conflict, not to specify exactly how to use them.

Reply:
I would say that the defence of caution with bans convinces me the other way round by being too weak — really, in this picture the _community_ loses nothing. I _do not want_ the community to try to spread values over the entire world, after all. I don't even want the community to have values, just respectful communication norms, in a sense.

I think if you want to just stress bans are on the table, this could be condensed a lot and would feel better. And no, the authority to ban is not the basis of _all_ soft power the team wields, if it is — we have a big problem. There is also actual respect to the effort invested into the project by the team members (including technical, be it code or documentation or testing).

«
While the team has to have the authority to ban permanently as the last resort, this should be applied only sparingly when less dractic measures are clearly inadequate for preserving well-being of the remaining community members.
»
seems to be enough to put permabans on the table, but also to remind to be careful with them (and to remind that permanence of a ban is not an absolute given).

(Vindictiveness… well, I have seen bans handed out for publically asking moderator A to ban moderator B because of a reasonable-in-context last warning being interpreted as a threat — in a sense you could say that the ensuing ban had a grain of vindictiveness, doesn't even matter if handed out by A or B).

Reply:
We can't have social norms if people can violate them, and continue violating them, and face no consequence. In fact, in the absence of the ability to ban, any hypothetical person whose goal was to get everyone fighting each other would have the strategic advantage over the nominal authority. NixOS is a very friendly community due to the leadership of its founders, and I admire and appreciate that, but divisive behavior is widespread in the world at large and we shouldn't imagine we're immune.

That's the sense in which the soft power is based on the hard power. I think we need to be honest with ourselves about that, trying to sugar-coat it would be a mistake.

Separately: I agree with the kind of respect that you're talking about here, and I share that value and I hope that it's implicit in the values we've set forth here. I want to take it as a chance to explain why we didn't list respect as one of these values. "Respect" means a lot of things to a lot of people, and some of those things are good and others of them are oppressive. It's very difficult to draw a distinction with that terminology and have it stick. I do think we captured the various *ideas* around respect, we just didn't use the word itself, in most places. That was a deliberate choice.

I appreciate you taking the time to write your proposed draft which tightens up the wording and removes the philosophical parts. I promise that we will consider it seriously. In the spirit of acknowledging my own position, I will say that I lean against removing the philosophy on bans because I think it's needed as part of anchoring the discussion in its concrete, real-world meaning. I would love to hear other people's perspectives on that.

Reply:
> That's the sense in which the soft power is based on the hard power. I think we need to be honest with ourselves about that, trying to sugar-coat it would be a mistake.

In reality, I think some people glide to ban without being noticeably affected by anything else, and others negotiate communication modes without anyone considering theoretical possibility of a ban as relevant, and the middle ground is quite rare. Sure, if 6-hour bans for cooling down heated topics come into fashion, it will be another story. Sure, there are exceptions. But I emphasized «all». There is a bit of soft power that comes together with hard power (but not that much). But you need actual soft power too, not just this forced fake, to ever be a mediator.

And the people who were banned were not paying enough attention to be divisive. The only division they sowed was the discussion if they were banned way too late (well, the discussion whether the ban was justified at all would definitely be more divisive…).

If you want philosophy to stay, then I think the justification why bans are needed should mention that some entities are not communicating, they are spamming; and some people insist on a communication style that a supermajority of people find undesirable to allow. Soft power is for those who communicate in good faith, hard power is for cleaning up — so I do not think soft power is relevant in the context of bans.

Then I would stress the community costs, not recognition of failure at preaching, as a reason why long-term bans are something to use as the last resort. Hopefully this menion of long-term shows that brief temporary bans are a different can of worms.

Then, after all that, indeed say that we expect that the efforts of the members of the team (including the efforts before joining the team) have earned them enough soft power to be able to resolve the tensions among real community members in a constructive way.

Reply:
I strongly favor keeping these sorts of things as simple as possible. We can retain the rationale in "comments" or in some other form.

Reply:
To be clear, our goal with the community team is to *develop* the types of leadership skills that you are referencing. They are important skills, and thus we want to teach them to everyone. As far as I can tell, I don't disagree with anything you're saying, and perhaps the "all" is a bit strong but I think I've clarified why it's there now?

Reply:
I think the text is not aiming to be long enough to be an explanation how to do things, I think the current version of the philosophical part is either wrong or uses weird choice of arguments in multiple places. So probably cheaper to drop than to rewrite, but if not drop, I have written above what I think should be recentered.

Reply:
Thanks. I do want to come back to these. I think it's closely connected to the points I've been going back and forth with @blaggacao over, elsewhere in this thread, about positive and negative ideals and trusting the community team to do its job. As I've said in several other comments, @ashkitten and I have not yet decided on where we stand on that, but it's got our attention.